File: src/main/java/tk/hiddenname/smarthome/exception/InvalidProcessingObjectTypeException.java
package tk.hiddenname.smarthome.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(value = HttpStatus.BAD_REQUEST)
public class InvalidProcessingObjectTypeException extends RuntimeException {

    public InvalidProcessingObjectTypeException() {
        super("Invalid ProcessingObject type");
    }
}

File: src/main/kotlin/tk/hiddenname/smarthome/exception/invalid/InvalidProcessingObjectTypeException.kt
package tk.hiddenname.smarthome.exception.invalid

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(value = HttpStatus.BAD_REQUEST)
class InvalidProcessingObjectTypeException(id: String)
    : RuntimeException("The type '$id'of processing object is invalid")